### ISSUE TITLE
[Bug] Incorrect Auto-rank on EduRec

### ISSUE THREAD

**Issue by chin-herng** [HollaG/tutreg#4]
First of all, I really liked this project! Kudos for making this possible and I would love to lend my help if time permits.

The following link did not export to the correct ranking on EduRec:

```
https://tutreg.com/order?share=CS3231:TUT:1,EL1101E:TUT:W3,GEN2050X:TUT:14,GEN2050X:TUT:13,EL1101E:TUT:W2,CS3231:TUT:2,CS3231:TUT:3,EL1101E:TUT:W1,GEN2050X:TUT:12,GEN2050X:TUT:04,EL1101E:TUT:W9,EL1101E:TUT:W11,GEN2050X:TUT:11,GEN2050X:TUT:06,EL1101E:TUT:W10,EL1101E:TUT:W7,GEN2050X:TUT:07,GEN2050X:TUT:08,EL1101E:TUT:W4,EL1101E:TUT:W5&classes=CS3231-TUT:1,2,3__EL1101E-TUT:W3,W2,W1,W9,W11,W10,W7,W4,W5__GEN2050X-TUT:14,13,12,04,11,06,07,08
```

The expected ranking as displayed on tutreg.com:

![image](https://github.com/user-attachments/assets/235e444a-e048-430b-950a-cdcabc76ad26)

When exported to EduRec, rank 8 appears duplicated:

![image](https://github.com/user-attachments/assets/16f56fae-531d-416d-8785-3b2e3b82ad0b)

EL1101E W1 is expected to be rank 8, however W10 and W11 are supposed to be rank 15 and 12 respectively. These ranks are missing on EduRec.

The rest of the rankings are correctly exported. Thus, it suffices to edit the rankings of W10 and W11 to the intended ones.

Do let me know if you are having trouble reproducing the issue on your side.

**Comment by HollaG** [comment]
Hi, thanks for the bug report. I've been able to isolate what I think is the problem, but I will not be able to test it right now, due to Chrome Extension Store's delayed rollout. I've attached a new build of the extension that should probably fix the issue, but I can't test as the courses I'm taking are not affected by this bug.

If you'd like to help me test, 
[build-v1.5-bugfix.zip](https://github.com/user-attachments/files/16595716/build-v1.5-bugfix.zip)

Download and unzip this file, 
then go to
`chrome://extensions`,
"Load Unpacked",
then click on the `build` folder after unzipping the bugfix zip,
then try your selection again. 

Thank you!
